pkg/server/bootstrap/api-export-identity_controller.go

```diff
@@ -0,0 +1,255 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package boostrap
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/wait"
+	coreinformers "k8s.io/client-go/informers/core/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/clusters"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+
+	configshard "github.com/kcp-dev/kcp/config/shard"
+	apisv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/apis/v1alpha1"
+	tenancyv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/tenancy/v1alpha1"
+	apisinformers "github.com/kcp-dev/kcp/pkg/client/informers/externalversions/apis/v1alpha1"
+	"github.com/kcp-dev/kcp/pkg/indexers"
+)
+
+const (
+	ControllerName = "kcp-api-export-identity-provider"
+	workKey        = "key"
+	configMapName  = "api-exports-identities"
+)
+
+// NewApiExportIdentityProviderController returns a new api export identity provider controller.
+//
+// The controller reconciles APIExports for the root APIs and
+// maintains a config map in the system:shard logical cluster
+// with identities per exports specified in the group or the group resources maps.
+//
+// The config map is meant to be used by clients/informers to inject the identities
+// for the given GRs when making requests to the server.
+func NewApiExportIdentityProviderController(
+	groupExportNames map[string]string,
+	groupResourceExportNames map[schema.GroupResource]string,
+	kubeClusterClient kubernetes.ClusterInterface,
+	remoteShardApiExportInformer apisinformers.APIExportInformer,
+	configMapInformer coreinformers.ConfigMapInformer,
+) (*controller, error) {
+	queue := workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), ControllerName)
+
+	c := &controller{
+		queue: queue,
+
+		kubeClusterClient: kubeClusterClient,
+
+		configMapIndexer:             configMapInformer.Informer().GetIndexer(),
```

Comment:
this might be expensive, all we really need is just a single configmap

Reply:
Is this the first instance of configmap watching in our code? IIRC APIExport already uses it, no>

Reply:
nope, I think that `installRootCAConfigMapController` brings the configmaps as well.